tokio/src/net/tcp/socket.rs

```diff
@@ -183,6 +183,127 @@ impl TcpSocket {
         self.inner.set_reuseaddr(reuseaddr)
     }
 
+    /// Retrieves the value set for `SO_REUSEADDR` on this socket
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use tokio::net::TcpSocket;
+    ///
+    /// use std::io;
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> io::Result<()> {
+    ///     let addr = "127.0.0.1:8080".parse().unwrap();
+    ///
+    ///     let socket = TcpSocket::new_v4()?;
+    ///     socket.set_reuseaddr(true)?;
+    ///     assert!(socket.get_reuseaddr().unwrap());
+    ///     socket.bind(addr)?;
+    ///
+    ///     let listener = socket.listen(1024)?;
+    /// # drop(listener);
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn get_reuseaddr(&self) -> io::Result<bool> {
+        self.inner.get_reuseaddr()
+    }
+
+    /// Allow the socket to bind to an in-use port. Only available for unix systems
+    /// (excluding Solaris).
+    ///
+    /// Behavior is platform specific. Refer to the target platform's
+    /// documentation for more details.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use tokio::net::TcpSocket;
+    ///
+    /// use std::io;
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> io::Result<()> {
+    ///     let addr = "127.0.0.1:8080".parse().unwrap();
+    ///
+    ///     let socket = TcpSocket::new_v4()?;
+    ///     socket.set_reuseport(true)?;
+    ///     socket.bind(addr)?;
+    ///
+    ///     let listener = socket.listen(1024)?;
+    /// # drop(listener);
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    #[cfg(all(unix, not(any(target_os = "solaris", target_os = "illumos"))))]
```

Comment:
```suggestion
    #[cfg(all(unix, not(target_os = "solaris"), not(target_os = "illumos")))]
    #[doc(cfg(all(unix, not(target_os = "solaris"), not(target_os = "illumos"))))]
```
![screenshot of set_reuseport](https://user-images.githubusercontent.com/928074/97807429-fb17a100-1c60-11eb-92d7-6d932b245084.png)

Reply:
#[doc(config)] is experimental so I've not added it.

Reply:
Ah, right. You need to do this to only conditionally enable it for docs.rs.
```rust
#[cfg_attr(docsrs, doc(cfg(all(unix, not(target_os = "solaris"), not(target_os = "illumos")))))]
```